### ISSUE TITLE
Existing Item Quantities at (default)Source set to zero when saving product images with gallery management

### ISSUE THREAD

**Issue by slacka1** [magento/inventory#2058]
### Preconditions

1. Magento 2.3
2. MSI 1.1.0
3. php 7.2
4. single store mode ("default" source only)
### Steps to reproduce

1.  create new simple product with ProductInterfaceFactory 
2. save with product repository (no stock data set yet)
3. save inventory to "default" source with:
 Magento\InventoryApi\Api\SourceItemsSaveInterface
4. inventory successfully saves
5. add images to product using:
'Magento\Catalog\Model\Product\Gallery\GalleryManagement'
6. Product inventory for "default" source, gets set to ZERO. 

This was super freaking tedious to debug.

If you save images FIRST with gallery management, and then set stock values after, stock quantities will not be set to 0 and things work as expected.

### Expected result

1. 

### Actual result

1. [Screenshots, logs or description]
2.

**Comment by IvanPletnyov** [comment]
@slacka1 Hello! If I understood correctly you are creating product, assign images and create stock items programmatically ? Thanks.

**Comment by slacka1** [comment]
Yeah, maybe it has to do with how i am adding the images. I'm not sure.
`if($bIncludesImageData){

	//remove existing images first:
	$arrMediaGalleryEntries = $objGalleryManagement->getList($csvdata_sSku);
	foreach($arrMediaGalleryEntries as $objMediaGalleryEntry){
		$iMediaGalleryEntryId = $objMediaGalleryEntry->getId();
		$objGalleryManagement->remove($csvdata_sSku, $iMediaGalleryEntryId);
	}

	//$csvdata_sImagesData = image_data;image_data;...
	$arrImages = explode(";",$csvdata_sImagesData);

	//process each image and it's options
	foreach($arrImages as $csvImageData){

		//$csvImageData = PATH::LABEL::BITFLAG_BASE::BITFLAG_SMALL::BITFLAG_THUMB::POS::BITFLAG_MOUSEOVER
		$arrImageData = explode("::", $csvImageData);

		if (isset($arrImageData[0]) && $arrImageData[0] !== ""){

			if($arrImageData[1] == ""){
				$arrImageData[1] = "ImageLabel";
			}

			//copy to base media folder
			$sDIRECTORY_PATH_media = $objectManager->get('Magento\Framework\App\Filesystem\DirectoryList')->getPath('media');
			//copies image to base media folder (your-magento-root-path/pub/media)
			$sIMG_PATH_URL = CopyImageToMediaFolder( $arrImageData[0], $objectManager, $sDIRECTORY_PATH_media );

			if(!$sIMG_PATH_URL == ""){

				//build image types array
				$arrImageTypes = array();
				if ($arrImageData[2] == "True"){
				array_push($arrImageTypes,"image");
				}
				if ($arrImageData[3] == "True"){
				array_push($arrImageTypes,"small_image");
				}
				if ($arrImageData[4] == "True"){	
				array_push($arrImageTypes,"thumbnail");
				}	

				//initialize Media Gallery Entry object
				unset($objMediaGalleryEntry);
				$objMediaGalleryEntry = $objectManager->create('\Magento\Catalog\Api\Data\ProductAttributeMediaGalleryEntryInterface');

				$objMediaGalleryEntry->setMediaType('image');
				$objMediaGalleryEntry->setTypes($arrImageTypes);
				$objMediaGalleryEntry->setFile($sIMG_PATH_URL);

				$objMediaGalleryEntry->setLabel($arrImageData[1]);
				$objMediaGalleryEntry->setPosition($arrImageData[5]);
				$objMediaGalleryEntry->setDisabled(false); //set true to prevent image from displaying in media gallery on product page.

				$fileData = file_get_contents($sIMG_PATH_URL);
				$imageType = exif_imagetype($sIMG_PATH_URL);
				$imageMimeType = image_type_to_mime_type($imageType);

				$imageData = base64_encode($fileData);
				$imageContent = $objectManager->create('\Magento\Framework\Api\Data\ImageContentInterface');
				//I was setting name to arrImageData[1] but it really doesn't like that. I think cuz name requires file extension?
				$imageContent->setName(basename($arrImageData[0]));
				$imageContent->setType($imageMimeType);
				$imageContent->setBase64EncodedData($imageData);
				$objMediaGalleryEntry->setContent($imageContent);

				$objGalleryManagement->create($objExistingProduct->getsku(), $objMediaGalleryEntry);

			}else{
				//log failed to add image.
			}
		}	
	} 
}`

**Comment by slacka1** [comment]
Here's how i set inventory. It works but if this is put before the gallery code above, inventory values are set to 0. Sorry, i don't know why my newlines aren't in effect.
`// MSI (1.1.0) - Multi-Source-Inventory:  @vendor/magento/module-inventory-api/ 				
$sSource = 'default'; //warehouse
$fQty    = $csvdata_dInvAvail;
$sSku    = $objExistingProduct->getSku();
$objSourceItemInterface = createSourceItemInterface($objectManager, $sSource, $fQty, $sSku);
$arrSourceItemInterfaces = array();
$arrSourceItemInterfaces[] = $objSourceItemInterface;
try{
	$objSourceItemsSaveInterface->execute($arrSourceItemInterfaces);	
}catch(Exception $e){
	echo "<br>Exception: $e<br>";
}`

**Comment by maghamed** [comment]
@IvanPletnyov @smoskaluk 
did we able to reproduce the issue or we need any other input from @slacka1 ?

**Comment by IvanPletnyov** [comment]
@slacka1 Hello, could you please create patch with your changes ?
